profile/src/main/resources/profileschema/datahelix.schema.json

```diff
@@ -180,29 +176,29 @@
           "type": "string"
         },
         "type": {
-          "type": "string",
-          "enum": [
-            "decimal",
-            "integer",
-            "string",
-            "datetime",
-            "date",
-            "time",
-            "ISIN",
-            "SEDOL",
-            "CUSIP",
-            "RIC",
-            "firstname",
-            "lastname",
-            "fullname",
-            "faker.firstname",
-            "faker.lastname",
-            "faker.name",
-            "faker.bloodgroup",
-            "faker.username",
-            "faker.nameprefix",
-            "faker.phonenumber",
-            "faker.cellnumber"
+          "oneOf" : [
+            {
+              "type": "string",
+              "enum": [
+                "decimal",
+                "integer",
+                "string",
+                "datetime",
+                "date",
+                "time",
+                "ISIN",
+                "SEDOL",
+                "CUSIP",
+                "RIC",
+                "firstname",
+                "lastname",
+                "fullname"
+              ]
+            },
+            {
+              "type": "string",
+              "pattern": "^faker\\.(\\w+.?)((\\w+).)*(\\w+)"
```

Comment:
Is this regex designed to accept any amount of `.` in the expression?

If so, is this present in the API (a cursory inspection didn't find anything that wasn't `faker.X.Y`)?

Reply:
As discussed, the current search space is very big and a lot of unpermissable inputs (ie. `faker.garbage.input`) could be put in. Hence, we've agreed that this is probably allowable and can be covered by the runtime error.